### PR TITLE
Format date w/ lang

### PR DESCRIPTION
**Pull request by stackmystack**

### CODE REVIEW

File: lib/arel_extensions/visitors/postgresql.rb [hunk 1]
```diff
@@ -8,9 +8,9 @@ class Arel::Visitors::PostgreSQL
 
       DATE_FORMAT_DIRECTIVES = {
         '%Y' => 'YYYY', '%C' => 'CC', '%y' => 'YY',
-        '%m' => 'MM', '%B' => 'FMMonth', '%^B' => 'FMMONTH', '%b' => 'Mon', '%^b' => 'MON',
```
Comment (akimd) [inline lib/arel_extensions/visitors/postgresql.rb]: Shouldn't this be squashed into the previous commit?

File: test/with_ar/all_agnostic_test.rb [hunk 2]
```diff
@@ -481,6 +481,46 @@ def test_format_date_with_names
...
+          'postgresql' => ->(l) { "SET lc_time to '#{l}';" }
+        }
+
+        User.connection.execute(sql[ENV['DB']][languages[ENV['DB']][lang]])
+      end
+
+      def test_format_date_with_names_and_lang_switch
+        skip "#{ENV['DB']} does not support word-based formatting for month and day names" if ['mssql'].include?(ENV['DB'])
+
+        # the bgin-rescue block is here to make sure we set the db back to en_US
```
Comment (akimd) [inline test/with_ar/all_agnostic_test.rb:501]: begin.

File: test/with_ar/all_agnostic_test.rb [hunk 3]
```diff
@@ -481,6 +481,46 @@ def test_format_date_with_names
...
+          assert_equal 'Monday, 03 March 14', t(@lucas, @updated_at.format('%A, %d %B %y'))
+          switch_to_lang(:fr)
+          if ['mysql'].include?(ENV['DB'])
+            assert_equal 'lun, 03 mar 14', t(@lucas, @updated_at.format('%a, %d %b %y'))
+            assert_equal 'lundi, 03 mars 14', t(@lucas, @updated_at.format('%A, %d %B %y'))
+          elsif ['postgresql'].include?(ENV['DB'])
+            assert_equal 'Lun., 03 Mars 14', t(@lucas, @updated_at.format('%a, %d %b %y'))
+            assert_equal 'Lundi, 03 Mars 14', t(@lucas, @updated_at.format('%A, %d %B %y'))
+          end
+          switch_to_lang(:en)
```
Comment (akimd) [inline test/with_ar/all_agnostic_test.rb:518]: I want to use `ensure`, not `rescue`.

**Review by akimd — CHANGES_REQUESTED** [review]

**Comment by stackmystack** [comment]
This whole PR should be discarded. There's no way we can guarantee consistent formatting by locale across different RDBMSes.

**Comment by akimd** [comment]
> This whole PR should be discarded. There's no way we can guarantee consistent formatting by locale across different RDBMSes.

Really?  I don't see a problem with at least the first commits.  You are "only" saying that we can't expect a single answer, but that's no reason, IMHO, to not support the formatting directives.  We just don't guarantee the result, but that's best effort.  The tests must be more tolerant I guess.

File: lib/arel_extensions/visitors/postgresql.rb (hunk 1 above)
Comment (stackmystack) [reply lib/arel_extensions/visitors/postgresql.rb]: It does

File: test/with_ar/all_agnostic_test.rb [hunk 4]
```diff
@@ -471,6 +471,61 @@ def test_format_iso_year_of_week
...
+        if ['mysql'].include?(ENV['DB'])
+          assert_equal 'lun, 03 mar 14', t(@lucas, @updated_at.format('%a, %d %b %y'))
+          assert_equal 'lundi, 03 mars 14', t(@lucas, @updated_at.format('%A, %d %B %y'))
+        elsif ['postgresql'].include?(ENV['DB'])
+          assert_equal 'Lun., 03 Mars 14', t(@lucas, @updated_at.format('%a, %d %b %y'))
+          assert_equal 'Lundi, 03 Mars 14', t(@lucas, @updated_at.format('%A, %d %B %y'))
+        elsif ['mssql'].include?(ENV['DB'])
+          assert_equal 'lundi, 03 mars 2014', t(@lucas, @updated_at.format('%A, %d %B %y'))
+        end
+        ensure
```
Comment (akimd) [inline test/with_ar/all_agnostic_test.rb:525]: ensure should be deindented.  Alined with def and end.

**Review by akimd — APPROVED** [review]

File: test/with_ar/all_agnostic_test.rb [hunk 5]
```diff
@@ -471,6 +471,61 @@ def test_format_iso_year_of_week
...
+      def test_format_date_with_names_and_lang_switch
+        skip "#{ENV['DB']} does not support word-based formatting for month and day names" if ['sqlite'].include?(ENV['DB'])
+
+        # the begin-rescue block is here to make sure we set the db back to en_US
+        # if we fail, so that other tests don't get contaminated.
+        #
+        # Tests should assert one single thing in principle, but until we
+        # refactor this whole thing, we'll have to do tricks of this sort.
+        switch_to_lang(:en)
+        if ['mysql', 'postgresql'].include?(ENV['DB'])
```
Comment (akimd) [inline test/with_ar/all_agnostic_test.rb:509]: btw (and sorry for being so slow at noticing, you may keep this for another time), this is a good ol' case.
```ruby
case ENV['DB']
when 'mysql', 'postgresql'
  assert_equal 'Mon, 03 Mar 14', t(@lucas, @updated_at.format('%a, %d %b %y'))
  assert_equal 'Monday, 03 March 14', t(@lucas, @updated_at.format('%A, %d %B %y'))
when 'mssql'
  ...
end
```

**Review by akimd — APPROVED** [review]
LGTM.

**Pull request merged**